lua/detour/util.lua

```diff
@@ -64,4 +72,42 @@ function M.overlap(window_a, window_b)
     return overlap_helper({M.get_text_area_dimensions(window_a)}, {M.get_text_area_dimensions(window_b)})
 end
 
+function M.find_top_popup()
+    local window_id = vim.api.nvim_get_current_win()
+    local all_covered_windows = vim.tbl_flatten(vim.tbl_values(internal.popup_to_covered_windows))
+    for popup, _ in pairs(internal.popup_to_covered_windows) do
+        if not vim.tbl_contains(all_covered_windows, popup) -- ignore popups with popups nested in them
+            and vim.tbl_contains(M.find_covered_bases(popup), window_id) then
```

Comment:
maybe we can replace this logic with using the lowest popup to `find_covered_bases` and then traversing the popup_to_covered_windows table to find the top popup from there. Not sure if it would make a difference but seems a bit safer.